test/unit/collection-view.filter.spec.js

```diff
@@ -0,0 +1,401 @@
+      this.collectionView = new this.CollectionView();
+      this.collectionView.render();
+      this.sinon.spy(this.collectionView, 'destroyEmptyView');
+    });
+
+    describe('when a model is added to the collection but rejected by the filter', function() {
+      beforeEach(function() {
+        this.collection.add(this.failModel);
+      });
+
+      it('should contain the empty view in the DOM', function() {
+        expect(this.collectionView.$el).to.contain.$text('empty');
+      });
+
+      it('should not destroy the empty view', function() {
+        expect(this.collectionView.destroyEmptyView).not.to.have.been.called;
+      });
+    });
+
+    describe('when a model is added to the collection and is accepted by the filter', function() {
+      beforeEach(function() {
+        this.collection.add(this.passModel);
+      });
+
+      it('should contain the empty view in the DOM', function() {
+        expect(this.collectionView.$el).to.contain.$text('true');
+      });
+
+      it('should destroy the empty view', function() {
+        expect(this.collectionView.destroyEmptyView).to.have.been.calledOnce
+          .and.calledOn(this.collectionView);
+      });
+    });
+  });
+
+  describe("combined with a sort", function() {
+    beforeEach(function() {
+      this.collection.comparator = 'bar';
+      this.model1 = new Backbone.Model({foo: true, bar: 1});
+      this.model2 = new Backbone.Model({foo: false, bar: 2});
+      this.model3 = new Backbone.Model({foo: true, bar: 3});
+      this.model4 = new Backbone.Model({foo: false, bar: 4});
+      this.collection.reset([this.model2, this.model4, this.model3, this.model1]);
+
+      this.ChildView = Marionette.ItemView.extend({
+        template: function(data) {
+          return data.bar;
+        }
+      });
+      this.collectionView = new this.CollectionView({childView: this.ChildView});
+    });
+
+    describe("when the filter does not change between sorts", function() {
+      beforeEach(function() {
+        this.collectionView.render();
+        this.collection.comparator = function(model) {
+          return -model.get('bar');
+        };
+        this.collection.sort();
+      });
+
+      it("only renders views that pass the filter", function() {
+        expect(this.collectionView.children.findByModel(this.model1)).to.exist;
+        expect(this.collectionView.children.findByModel(this.model2)).not.to.exist;
+        expect(this.collectionView.children.findByModel(this.model3)).to.exist;
+        expect(this.collectionView.children.findByModel(this.model4)).not.to.exist;
+      });
+
+      it("renders the views in the correct order", function() {
+        expect(this.collectionView.$el).to.contain.$text('31');
+      });
+    });
+
+    describe("when the filter changes between sorts", function() {
+      beforeEach(function() {
+        this.collectionView.render();
+        this.collectionView.filter = this.inverseFilter;
+        this.collection.sort();
```

Comment:
In this case, I want to make sure that only changing the filter will cause the view to re-render. The expectation is that a collectionview will update on a sort, and since the filter can possibly depend on the index, we want that to cause a render